### PR TITLE
🐛 Ignore groups not served  by the cluster

### PR DESCRIPTION
**Pull request by jerr**
##### Description

If a group is not available on a server the `discovery.ServerPreferredResources` function returns  the list of preferred resources but also an [`ErrGroupDiscoveryFailed`](https://github.com/kubernetes/client-go/blob/v0.24.1/discovery/discovery_client.go#L331) error with the groups not available. 

This error can be ignored  because the unavailable resource types  should not be audited.

#
This is an example of a cluster with the `external.metrics.k8s.io` group not available:
```
$ kubectl  get apiservice v1beta1.external.metrics.k8s.io 
NAME                              SERVICE                    AVAILABLE                      AGE
v1beta1.external.metrics.k8s.io   my-custom-metrics-server   False (FailedDiscoveryCheck)   22d
```
```
$ kubectl  get --raw /apis/external.metrics.k8s.io/v1beta1
Error from server (ServiceUnavailable): the server is currently unable to handle the request
```
If we have a cluster in this state no resources are audited because an error is returned by ` discovery.ServerPreferredResources`
https://github.com/Shopify/kubeaudit/blob/6a30cc08c46cee8922f451b0449923b7a3ec761b/internal/k8sinternal/client.go#L129-L130

#
##### Type of change

- [x] Bug fix :bug:

##### How Has This Been Tested?

- [x] Automated tests still work
- [x] Manual tests: The `kubeaudit all` command was tested on a cluster with an unavailable  `external.metrics.k8s.io` group. Without this fix no reports are displayed,  with the fix the reports are provided as expected.
  
##### Checklist:

- [x] I have :tophat: my changes (A 🎩 specifically includes pulling down changes, setting them up, and manually testing the changed features and potential side effects to make sure nothing is broken)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage did not decrease
- [x] I have signed the appropriate [Contributor License Agreement](https://cla.shopify.com/)

### CODE REVIEW

File: internal/k8sinternal/client.go [hunk 1]
```diff
@@ -207,5 +210,10 @@ func (kc kubeClient) GetKubernetesVersion() (*version.Info, error) {
 
 // ServerPreferredResources returns the supported resources with the version preferred by the server.
 func (kc kubeClient) ServerPreferredResources() ([]*metav1.APIResourceList, error) {
-	return discovery.ServerPreferredResources(kc.discoveryClient)
+	list, err := discovery.ServerPreferredResources(kc.discoveryClient)
+	// If a group is not served by the cluster the resources of this group will not be audited.
+	if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); ok {
```
Comment (genevieveluyt) [inline internal/k8sinternal/client.go:215]: Can we use `errors.Is`?

**Review by genevieveluyt — APPROVED** [review]
Thanks! Just an optional nit

File: internal/k8sinternal/client.go (hunk 1 above)
Comment (jerr) [reply internal/k8sinternal/client.go:215]: > Can we use `errors.Is`?

`errors.Is` needs an error instance right? We want to check its type here.

Comment (genevieveluyt) [reply internal/k8sinternal/client.go:215]: `errors.As` then? 😄

**Review by genevieveluyt — APPROVED** [review]

**Pull request merged**